### ISSUE TITLE
[Issue]: n2 elite operations stuck at Mifare Ultralight scanning

### ISSUE THREAD

**Issue by ludeviance** [HiddenRamblings/TagMo#672]
### Issue Description

Every operation I try to make on the n2 elite tab (write to one or many slots, set active slot, delete slot) get stuck at the tag reading screen with the message "mifare ultralight scanning..."

![Screenshot_2023-05-01-11-47-44-963_com hiddenramblings tagmo eightbit](https://user-images.githubusercontent.com/77011580/235472222-cb9bf8b2-2728-4a7a-817c-6bc131bcefde.jpg)

If I find an amiibo image from the browser tag and write from there, after I select the slot number and place the n2 behind my phone, I get the same screen, but with the message "N2 elite scanning..." and after less than a second, the operation is completed successfully. That way, I can only write on one slot and nothing more. 

### Logcat Output

```shell
improperly called by android.widget.LinearLayout{25d3b4d V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{5543486 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{27f6147 V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{6438705 V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{3ca3d74 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{78db29d V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{8c091bd V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{62f1312 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{bc6cae3 V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.824 15587 15616 W Parcel  : Expecting binder but got null!
```

### Usage Requirements

- [X] I have read and understood the usage requirements for TagMo

**Comment by AbandonedCart** [comment]
What happened to the rest of the log?

**Comment by ludeviance** [comment]
Sorry, I think I miss-copied the log. Here it is:

```shell
TagMo 3.7.5 (GitHub Release) #3fa2080
Xiaomi TIRAMISU (13) - 10.00 GB RAM

--------- beginning of system
05-01 11:49:11.279 15587 15642 W BpBinder: Slow Binder: BpBinder transact took 478 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:11.280 15587 15646 W BpBinder: Slow Binder: BpBinder transact took 482 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:11.571 15587 15642 W BpBinder: Slow Binder: BpBinder transact took 291 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:11.700 15587 15647 W BpBinder: Slow Binder: BpBinder transact took 904 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:11.980 15587 15645 W BpBinder: Slow Binder: BpBinder transact took 1179 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.106 15587 15646 W BpBinder: Slow Binder: BpBinder transact took 825 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.287 15587 15649 W BpBinder: Slow Binder: BpBinder transact took 1479 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.362 15587 15642 W BpBinder: Slow Binder: BpBinder transact took 790 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.605 15587 15643 W BpBinder: Slow Binder: BpBinder transact took 1796 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.650 15587 15647 W BpBinder: Slow Binder: BpBinder transact took 950 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
--------- beginning of main
05-01 11:49:12.766 15587 15587 W MirrorManager: this model don't Support
05-01 11:49:12.874 15587 15587 W BottomSheetBehavior: Cannot set state: 5
05-01 11:49:12.911 15587 15644 W BpBinder: Slow Binder: BpBinder transact took 2108 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.912 15587 15645 W BpBinder: Slow Binder: BpBinder transact took 933 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:12.927 15587 15587 W libc    : Access denied finding property "[ro.vendor.oplus.market.name](http://ro.vendor.oplus.market.name/)"
05-01 11:49:12.927 15587 15587 W libc    : Access denied finding property "[ro.vendor.oplus.market.name](http://ro.vendor.oplus.market.name/)"
05-01 11:49:12.927 15587 15587 W libc    : Access denied finding property "[ro.vendor.oplus.market.name](http://ro.vendor.oplus.market.name/)"
05-01 11:49:12.965 15587 15587 W ziparchive: Unable to open '/data/app/~~wvs4Q3uKMFsK-ZvHP3HqiQ==/com.google.android.trichromelibrary_561513534-x6zZCsoY7Z-lhoY9-0Pjxw==/[base.dm](http://base.dm/)': No such file or directory
05-01 11:49:12.965 15587 15587 W ziparchive: Unable to open '/data/app/~~wvs4Q3uKMFsK-ZvHP3HqiQ==/com.google.android.trichromelibrary_561513534-x6zZCsoY7Z-lhoY9-0Pjxw==/[base.dm](http://base.dm/)': No such file or directory
05-01 11:49:12.965 15587 15587 W .tagmo.eightbit: Entry not found
05-01 11:49:13.087 15587 15699 W cr_media: BLUETOOTH_CONNECT permission is missing.
05-01 11:49:13.088 15587 15699 W cr_media: registerBluetoothIntentsIfNeeded: Requires BLUETOOTH permission
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/alac
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/x-ape
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/evrc
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/qcelp
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/dsd
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/amr-wb-plus
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/x-ms-wma
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/evrc
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/qcelp
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/alac
05-01 11:49:13.152 15587 15706 W AudioCapabilities: Unsupported mime audio/x-ape
05-01 11:49:13.155 15587 15706 W VideoCapabilities: Unsupported mime image/vnd.android.heic
05-01 11:49:13.155 15587 15706 W VideoCapabilities: Unsupported mime image/vnd.android.heic
05-01 11:49:13.155 15587 15706 W VideoCapabilities: Unsupported mime video/dolby-vision
05-01 11:49:13.155 15587 15706 W VideoCapabilities: Unsupported mime video/dolby-vision
05-01 11:49:13.182 15587 15648 W BpBinder: Slow Binder: BpBinder transact took 2373 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:13.334 15587 15646 W BpBinder: Slow Binder: BpBinder transact took 1227 ms, interface=com.android.vending.billing.IInAppBillingService, code=9 oneway=false
05-01 11:49:17.535 15587 15616 W Parcel  : Expecting binder but got null!
05-01 11:49:18.824 15587 15616 W Parcel  : Expecting binder but got null!
05-01 11:49:22.329 15587 15587 W View    : requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{63e576f V.ED..... ......ID 189,38-374,109} during layout: running second layout pass
05-01 11:49:22.329 15587 15587 W View    : requestLayout() improperly called by androidx.recyclerview.widget.RecyclerView{6093c7c VFED.V... ......ID 0,232-1080,2093 #7f0900dc app:id/elite_content} during layout: running second layout pass
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{b0a9f14 V.E..V... ......ID 0,0-1026,200} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{a857622 V.E..V... ........ 0,62-805,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{51dd8b3 V.E..V... ........ 0,98-805,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{a52bcac V.E..V... ........ 0,0-1026,200} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{32dfe70 V.E..V... ........ 0,62-805,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{8bba5e9 V.E..V... ........ 0,98-805,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{89df544 V.E..V... ........ 0,0-1026,200} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{fa8d26e V.E..V... ........ 0,62-805,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{a134d0f V.E..V... ........ 0,98-805,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{9d4a8dc V.E..V... ........ 0,0-1026,200} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{44c919c V.E..V... ........ 0,62-805,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:22.385 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{39fb7a5 V.E..V... ........ 0,98-805,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:27.047 15587 15601 W .tagmo.eightbit: Cleared Reference was only reachable from finalizer (only reported once)
05-01 11:49:27.059 15587 15616 W Parcel  : Expecting binder but got null!
05-01 11:49:29.661 15587 15814 W IllegalStateException: java.lang.IllegalStateException: Call connect() first!
05-01 11:49:29.661 15587 15814 W IllegalStateException: at android.nfc.tech.BasicTagTechnology.checkConnected(BasicTagTechnology.java:52)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at android.nfc.tech.BasicTagTechnology.transceive(BasicTagTechnology.java:146)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at android.nfc.tech.NfcA.transceive(NfcA.java:129)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NTAG215.transceive(NTAG215.kt:65)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NTAG215.readSignature(NTAG215.kt:121)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.TagArray.isElite(TagArray.kt:83)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NfcActivity.onTagDiscovered(NfcActivity.kt:232)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NfcActivity.onNewIntent$lambda$43(NfcActivity.kt:699)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NfcActivity.$r8$lambda$92RVe9oothvWoGBggZAH5myOJhY(NfcActivity.kt:0)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at com.hiddenramblings.tagmo.nfctech.NfcActivity$$ExternalSyntheticLambda5.run(R8$$SyntheticClass:0)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
05-01 11:49:29.661 15587 15814 W IllegalStateException: at java.lang.Thread.run(Thread.java:1012)
05-01 11:49:49.308 15587 15587 W View    : requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{63e576f V.ED..... ......ID 189,38-353,109} during layout: running second layout pass
05-01 11:49:49.308 15587 15587 W View    : requestLayout() improperly called by androidx.recyclerview.widget.RecyclerView{acae9eb VFED.V... ......ID 0,0-1080,0 #7f090108 app:id/foomiibo_list} during layout: running second layout pass
05-01 11:49:49.308 15587 15587 W View    : requestLayout() improperly called by androidx.recyclerview.widget.RecyclerView{26d6348 VFED.V... ......ID 0,0-1080,2109 #7f090080 app:id/browser_content} during layout: running second layout pass
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{39d60dd V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{a3666ba V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{a52996b V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{d90ce95 V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{3ac1cc8 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{bbd6161 V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{25d3b4d V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{5543486 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{27f6147 V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{6438705 V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{3ca3d74 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{78db29d V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{8c091bd V.E..V... ........ 0,0-1048,217} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{62f1312 V.E..V... ........ 0,62-827,98 #7f090116 app:id/group1} during second layout pass: posting in next frame
05-01 11:49:49.362 15587 15587 W View    : requestLayout() improperly called by android.widget.LinearLayout{bc6cae3 V.E..V... ........ 0,98-827,134 #7f090117 app:id/group2} during second layout pass: posting in next frame
05-01 11:49:49.824 15587 15616 W Parcel  : Expecting binder but got null!
```

**Comment by AbandonedCart** [comment]
Fixed by 1a47e2acabdb3777108c23cb5c3eed281653df57
